c2corg_ui/static/js/documentediting.js

```diff
@@ -595,6 +594,39 @@ app.DocumentEditingController.prototype.updateMaxSteps = function(waypointType)
 
 
 /**
+ * Check what properties are missing and tell the user.
+ * @param {Object} doc
+ * @export
+ */
+app.DocumentEditingController.prototype.checkMissingProps = function(doc) {
+  var type = doc.type ? app.utils.getDoctype(doc.type) : this.modelName_ + 's'; // missing 's' in modelName if creating a new document
+  var fields = app.constants.REQUIRED_FIELDS[type];
+  var missing = {};
+  var error = [];
+
+  for (var i = 0; i < fields.length; i++) {
+    if (fields[i] === 'title' || fields[i] === 'lang') {
```

Comment:
All those ``fields[i]`` make the code hard to read. You should use an intermediate ``var field = fields[i];``.